assets/js/modules/reader-revenue-manager/datastore/publications.js

```diff
@@ -45,6 +49,98 @@ const baseInitialState = {
 };
 
 const baseActions = {
+	/**
+	 * Syncronizes the onboarding state of the publication with the API.
+	 * Updates the settings on the server.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {void}
+	 */
+	*syncPublicationOnboardingState() {
+		const registry = yield commonActions.getRegistry();
+		const connected = yield commonActions.await(
+			registry
+				.resolveSelect( CORE_MODULES )
+				.isModuleConnected( MODULE_SLUG )
+		);
+
+		// If the module is not connected, do not attempt to sync the onboarding state.
+		if ( ! connected ) {
+			return;
+		}
+
+		yield commonActions.await(
+			registry
+				.resolveSelect( MODULES_READER_REVENUE_MANAGER )
+				.getSettings()
+		);
+
+		const publicationID = registry
+			.select( MODULES_READER_REVENUE_MANAGER )
+			.getPublicationID();
+
+		// If there is no publication ID, do not attempt to sync the onboarding state.
+		if ( publicationID === undefined ) {
+			return;
+		}
+
+		const publications = registry
+			.select( MODULES_READER_REVENUE_MANAGER )
+			.getPublications();
+
+		// If there are no publications, do not attempt to sync the onboarding state.
+		if ( ! publications ) {
+			return;
+		}
+
+		const publication = publications.find(
+			// eslint-disable-next-line sitekit/acronym-case
+			( { publicationId } ) => publicationId === publicationID
+		);
+
+		// If the publication is not found, do not attempt to sync the onboarding state.
+		if ( ! publication ) {
+			return;
+		}
+
+		const { onboardingState } = publication;
+		const currentOnboardingState = registry
+			.select( MODULES_READER_REVENUE_MANAGER )
+			.getPublicationOnboardingState();
+
+		const settings = registry
+			.select( MODULES_READER_REVENUE_MANAGER )
+			.getSettings();
+
+		if ( onboardingState !== currentOnboardingState ) {
+			settings.publicationOnboardingState = onboardingState;
+		}
+
+		// eslint-disable-next-line sitekit/no-direct-date
+		settings.publicationOnboardingStateLastSyncedAtMs = Date.now();
+
+		registry
+			.dispatch( MODULES_READER_REVENUE_MANAGER )
+			.setSettings( settings );
+
+		// Save the settings to the API.
+		registry.dispatch( MODULES_READER_REVENUE_MANAGER ).saveSettings();
+
+		// If the onboarding state is complete, set the key in CORE_UI to trigger the notification.
+		if (
+			onboardingState ===
+			PUBLICATION_ONBOARDING_STATES.ONBOARDING_COMPLETE
+		) {
+			registry
+				.dispatch( CORE_UI )
+				.setValue(
+					UI_KEY_READER_REVENUE_MANAGER_SHOW_PUBLICATION_APPROVED_NOTIFICATION,
+					true
+				);
+		}
```

Comment:
I think the idea here was to set this only if the onboarding state changed from something else to `ONBOARDING_COMPLETE`. With the current implementation, the [`PublicationApprovedOverlayNotification`](https://github.com/google/site-kit-wp/issues/8843) will be shown even if the user sets up RRM with a publication that had onboarding completed to start with.

@ankitrox Any inputs?

CC: @techanvil

Reply:
Thanks for spotting this, @nfmohit, looks like we simply missed this one. I've created a followup issue to address it, https://github.com/google/site-kit-wp/issues/9089. I've left it with you in ACR, please take a look.

Reply:
Thanks for spotting this @nfmohit . I've created a follow-up PR for this issue https://github.com/google/site-kit-wp/pull/9090

Reply:
Hey @ankitrox, thanks for raising the PR. However, with this issue now being closed, we'll need to create the fix against the newly created issue #9089.

Reply:
Thanks @techanvil 

I will keep that PR as is and we can link that PR to the issue when it is up for execution.

Reply:
Thank you for sorting this out, folks!